sys/net/gnrc/network_layer/ipv6/util.c

```diff
@@ -0,0 +1,133 @@
+        ctx = NULL;
+    }
+    else {
+        cid = ctx->flags_id & GNRC_SIXLOWPAN_CTX_FLAGS_CID_MASK;
+    }
+    /* find first free context ID */
+    if (ctx == NULL) {
+        while (((ctx = gnrc_sixlowpan_ctx_lookup_id(cid)) != NULL) &&
+               !_ctx_match(ctx, prefix, prefix_len)) {
+            cid++;
+        }
+    }
+    if (cid < GNRC_SIXLOWPAN_CTX_SIZE) {
+        DEBUG("GNRC util: add compression context %u for prefix %s/%u\n", cid,
+              ipv6_addr_to_str(addr_str, prefix, sizeof(addr_str)),
+              prefix_len);
+        gnrc_sixlowpan_ctx_update(cid, (ipv6_addr_t *)prefix, prefix_len,
+                                  valid / (60 * MS_PER_SEC),
+                                  true);
+    }
+}
+
+void gnrc_util_conf_prefix(gnrc_netif_t *netif, const ipv6_addr_t *pfx,
+                           unsigned pfx_len, uint32_t valid, uint32_t pref,
+                           bool tentative)
+{
+    eui64_t iid;
+    ipv6_addr_t addr;
+    uint8_t flags = tentative
+                  ? (GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_TENTATIVE & 0x1)
+                  : GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID;
+
+    assert(netif != NULL);
+    DEBUG("GNRC util: (re-)configure prefix %s/%d\n",
+          ipv6_addr_to_str(addr_str, pfx, sizeof(addr_str)), pfx_len);
+    if (gnrc_netapi_get(netif->pid, NETOPT_IPV6_IID, 0, &iid,
+                        sizeof(eui64_t)) >= 0) {
+        ipv6_addr_set_aiid(&addr, iid.uint8);
+    }
+    else {
+        LOG_WARNING("GNRC util: cannot get IID of netif %u\n", netif->pid);
+        return;
+    }
+    ipv6_addr_init_prefix(&addr, pfx, pfx_len);
+    /* add address as tentative */
+    if (gnrc_netif_ipv6_addr_add_internal(netif, &addr, pfx_len, flags) > 0) {
+        /* update lifetime */
+        if (valid < UINT32_MAX) { /* UINT32_MAX means infinite lifetime */
+            /* the valid lifetime is given in seconds, but the NIB's timers work
+             * in microseconds, so we have to scale down to the smallest
+             * possible value (UINT32_MAX - 1). */
+            valid = (valid > (UINT32_MAX / MS_PER_SEC)) ?
+                          (UINT32_MAX - 1) : valid * MS_PER_SEC;
+        }
+        if (pref < UINT32_MAX) { /* UINT32_MAX means infinite lifetime */
+            /* same treatment for pref */
+            pref = (pref > (UINT32_MAX / MS_PER_SEC)) ?
+                         (UINT32_MAX - 1) : pref * MS_PER_SEC;
+        }
+        gnrc_ipv6_nib_pl_set(netif->pid, pfx, pfx_len, valid, pref);
+        if (IS_USED(MODULE_GNRC_IPV6_NIB) &&
+            IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LBR) &&
+            IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C) &&
+            gnrc_netif_is_6ln(netif)) {
+            if (IS_USED(MODULE_GNRC_SIXLOWPAN_CTX)) {
+                _update_6ctx(pfx, pfx_len, valid);
+            }
+            (void)gnrc_ipv6_nib_abr_add(&addr);
+        }
+        if (IS_USED(MODULE_GNRC_RPL)) {
```

Comment:
So
```suggestion
        if (IS_USED(MODULE_GNRC_RPL) &&
            IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LBR)) {
```
?

Reply:
A border router does not necessarily has to be a RPL root or a RPL root must not be a border router. These lines purely exist to make our `gnrc_border_router` example happy (where the border router is assumed to be a RPL root), so this should not be here, but in a 6LBR-specific module (like it is the case with DHCPv6 and with ethos somewhat, because ethos is our thing anyways).

Reply:
Yes, I think this can be a nice convenience function within the `gnrc_rpl` function. The `rpl root` shell command probably can also benefit from that.